### PR TITLE
Single quotes and empty attributes

### PR DESCRIPTION
**Pull request by kaltsimon**
Hi,
I was converting some Angular templates to pug and noticed that if there are single quotes in attribute values the result is invalid, e.g. `<h2 class="'x'"></h2> --> h2(class=''x'')`. I changed this so the parser outputs double quotes for these attributes, resulting in `h2(class="'x'")`.

I also thought I'd leave out attribute values that are empty: `span(translate)` instead of `span(translate='')`.

### CODE REVIEW

**Review by nonsocode — APPROVED** [review]

**Review by KenRoda — APPROVED** [review]

**Comment by izolate** [comment]
Hi @kaltsimon, I'm sorry for the late reply. Nice work over here, thank you.

I think the easiest approach is adding an escaped single quote in attribute values, as I've done in this commit (https://github.com/izolate/html2pug/commit/1d03c7b71245f6bb590b2f180e3f9335438314ea). 

This would transform the HTML:
```html
<button onclick="javascript:alert('foo')">alert</button>
```

to the following valid Pug:
```pug
button(onclick='javascript:alert(\'foo\')') alert
``` 

What do you think?

**Comment by izolate** [comment]
Hi @kaltsimon, I'm sorry for the late reply. Nice work over here, thank you.

I think the easiest approach is adding an escaped single quote in attribute values, as I've done in this commit (https://github.com/izolate/html2pug/commit/1d03c7b71245f6bb590b2f180e3f9335438314ea). 

This would transform the HTML:
```html
<button onclick="javascript:alert('foo')">alert</button>
```

to the following valid Pug:
```pug
button(onclick='javascript:alert(\'foo\')') alert
```

**Comment by izolate** [comment]
Hi @kaltsimon, I'm sorry for the late reply. Nice work over here, thank you.

I think the easiest approach is adding an escaped single quote in attribute values, as I've done in this commit (https://github.com/izolate/html2pug/commit/1d03c7b71245f6bb590b2f180e3f9335438314ea). 

This would transform the HTML:
```html
<button onclick="javascript:alert('foo')">alert</button>
```

to the following valid Pug:
```pug
button(onclick='javascript:alert(\'foo\')') alert
```

File: src/parser.js [hunk 1]
```diff
@@ -108,7 +108,12 @@ class Parser {
           pugNode += `.${value.split(' ').join('.')}`
           break
         default:
-          attributeList.push(`${name}='${value}'`)
+          if (!value) {
+            attributeList.push(`${name}`)
```
Comment (izolate) [inline src/parser.js:112]: This is great! If you could rebase with `develop` branch and include only this conditional, I'd be happy to merge it in

Comment (izolate) [reply src/parser.js:112]: Added in 075d462cd0d104e7f8dfa98e231213b1ea6b541d

**Comment by izolate** [comment]
Closing in favour of #35.

Thank you for your contributions. The library core has been refactored and these issues addressed in the linked PR.

**Pull request closed without merging**